src/main/java/entities/Post.java

```diff
@@ -0,0 +1,168 @@
+package entities;
+import java.util.Date;
+import java.util.List;
+
+/**
+ * The Post class contains all the necessary information of a post: its ID, title,
+ * description,tags,collaborators, deadline, and users who favourited it.
+ */
+public class Post extends Postable{
+    public static int numPostsCreated = 0;
+    /**The id is equal to the number of posts created.*/
+    private int id;
+    /**This is a list of the users who favourited this post.*/
+    private List<User> favouritedUsers;
+    private String title;
+    private List<String> tags;
+    private String collaborators;
+    private Date deadline;
+
+    /**
+     * @param poster the User who is posting this Post. This is an attribute of the superclass Postable.
+     * @param title this Post's title.
+     * @param mainDesc this Post's description.
+     * @param tags the tags this Post is added to.
+     * @param collaborators a description of the type of the collaborators the User who posted this Posts is looking for.
+     * @param deadline the date on which the Post is to be deleted from the database.
+     * @param numPostsCreated1 the number of posts created so far.
+     */
+    public Post(User poster, String title, String mainDesc, List<String> tags, String collaborators,
+                Date deadline, int numPostsCreated1){
+        super.user = poster;
+        super.body = mainDesc;
+        numPostsCreated = numPostsCreated1;
+        numPostsCreated++;
+        this.id = numPostsCreated;
+        this.title = title;
+        this.tags = tags;
+        this.collaborators = collaborators;
+        this.deadline = deadline;
+    }
+
+    /**
+     * @return the Post's ID.
+     */
+    public int getID(){
+        return this.id;
+    }
+
+    /**
+     * @return a list of the Users who favourited the Post.
+     */
+    public List<User> getFavouritedUsers(){
+        return this.favouritedUsers;
+    }
+
+    /**
+     * @return a list of tags specified by the User who made the Post.
+     */
+    public List<String> getTags(){
+        return this.tags;
+    }
+
+    /**
+     *
+     * @return the number of Users who favourited the Post.
+     */
+    public int getNumFavouritedUsers(){
+        return this.favouritedUsers.size();
+    }
+
+    /**
+     * @return this post's deadline.
+     */
+    public Date getDeadline(){
+        return this.deadline;
+    }
+
+    /**
+     * @return the replies made to this Post.
+     */
+    public List<Postable> getReplies(){
```

Comment:
Might need to change return type to List<Comment>, the autograding says it can't convert List<Comment> to List<Postable>

Reply:
Can you clarify? I thought the return type is List already

Reply:
Sorry, I don't know why it ignored the stuff inside the chevrons. It should say change List\<Postable\> to List\<Comment\>

Reply:
I think the problem is that, currently, the type for replies is List<Comment> but when the new PR for comment is approved, it will be List<Postable> and then it will work.

Reply:
Ah yes, just checked the other PR. Everything is fine then